commit 6cf64f039140e394906ae10c544eca77dd7d8e3f (refs/changes/32/5032/5)
Author: Robert Bogart <Robert.bogart@joyent.com>
Date:   2018-11-15T22:45:47+00:00 (11 months ago)
    
    MANTA-4015 Add mako datacenter to header information when uploading mako manifest
    Reviewed by: David Pacheco <dap@joyent.com>
    Approved by: Jan Wyszynski <jan.wyszynski@joyent.com>

diff --git a/bin/upload_mako_ls.sh b/bin/upload_mako_ls.sh
index 9ea52bc..dd10c4e 100755
--- a/bin/upload_mako_ls.sh
+++ b/bin/upload_mako_ls.sh
@@ -44,15 +44,11 @@ MAKO_DIR=/opt/smartdc/mako
 TARGET_DIR=/manta
 START_TIME=`date -u +"%Y-%m-%dT%H:%M:%SZ"` # Time that this script started.
 
-
-
 # Mutables
 
 NOW=""
 SIGNATURE=""
 
-
-
 ## Functions
 
 function fatal {
@@ -61,13 +57,11 @@ function fatal {
     exit 1
 }
 
-
 function log {
     local LNOW=`date`
     echo "$LNOW: $(basename $0): info: $*" >&2
 }
 
-
 function sign() {
     NOW=$(date -u "+%a, %d %h %Y %H:%M:%S GMT")
     SIGNATURE=$(echo "date: $NOW" | tr -d '\n' | \
@@ -76,7 +70,6 @@ function sign() {
         || fatal "unable to sign data"
 }
 
-
 function manta_put_directory() {
     sign || fatal "unable to sign"
     curl -fsSk \
@@ -88,14 +81,21 @@ function manta_put_directory() {
         $MANTA_URL/$MANTA_USER/stor/${1} 2>&1
 }
 
-
 function manta_put() {
     sign || fatal "unable to sign"
+
+    datacenter=$(mdata-get sdc:datacenter_name)
+
+    if [[ -z "$datacenter" ]]; then
+        fatal "unable to determine datacenter"
+    fi
+
     curl -vfsSk \
         -X PUT \
         -H "Date: $NOW" \
         -H "Authorization: Signature $AUTHZ_HEADER,signature=\"$SIGNATURE\"" \
         -H "Connection: close" \
+        -H "m-datacenter: $datacenter" \
         -H "m-mako-dump-time: $START_TIME" \
         $MANTA_URL/$MANTA_USER/stor/${1} \
         -T $2 \
